### DISCUSSION TITLE
Flowsie to WhatsApp Chatbot

### DISCUSSION THREAD

**Discussion by christina0w0** [FlowiseAI/Flowise#932] [Q&A]
Hey, can the bot built on Flowise be linked to WhatsApp chatbot?

**Comment by HenryHengZJ** [comment]
yes, most people use Flowise API and connect to apps like Zapier/n8n to achieve that

**Reply by rudiranck** [reply]
How? Did not find Flowise and Whatsapp integration on any of those

**Reply by ernimen7** [reply]
Curl request.

**Reply by suyogwaghere** [reply]
How to integrate Flowise API and WhatsApp Chatbot?

**Reply by Rudi-AI-Oficial** [reply]
Thanks @ernimen7 ! It works🎉
Now the problem is, every time the API through Curl request is called it forgets what has been said (it is recreating the OpenAI Thread) How to work with Curl request while keeping the thread. I am calling Flowise with Make. I guess I need to create a flow on flowise such a way to get the token as soon is created and force it to be input back. how...

**Reply by HenryHengZJ** [reply]
You need to specify chatId https://docs.flowiseai.com/integrations/memory/threads

**Reply by Rudi-AI-Oficial** [reply]
@HenryHengZJ  I did it, I can see a chatId included in the Json output, but only stays there, and flowise is still generating a new threadId!
![Screenshot 2024-01-10 125619](https://github.com/FlowiseAI/Flowise/assets/155680808/1757dff9-b540-47b1-8be5-706938a7924d)

**Reply by Rudi-AI-Oficial** [reply]
I noticed that if I call the API using python it works.... it keeps the threadID for the same userID.
But something goes wrong when calling through Make.com
